lib/hobbes/lang/pat/dfa.C

```diff
@@ -1267,9 +1268,23 @@ ExprPtr liftDFAExprWithSwitchCompression(MDFA* dfa, stateidx_t state) {
 bool shouldInlineState(const MDFA* dfa, stateidx_t state) {
   const MStatePtr& s = dfa->states[state];
 
+  // stop adding state related IR code into current function, if the state value
+  // is above `HOBBES_DFA_INLINE_THRESHOLD`. If a DFA has too many states,
+  // otherwise IR function can get so large that llvm cannot handle
+  static const auto inlineThreshold = [] {
+    // setting it to 0 makes no threshold. It must be a non-negative integer
+    const char* v = std::getenv("HOBBES_DFA_INLINE_THRESHOLD");
+    if (v == nullptr) {
+      return 2'000UL; // empirical data
+    }
+    static_assert(std::is_same<stateidx_t, unsigned long>::value, "");
```

Comment:
Either add a descriptive message, or (if C++17 is available) don't add one at all. Now the developer gets no usable information in the error message if the static assertion fails (worst of both worlds).